src/components/MainApp.js

```diff
@@ -7,42 +7,56 @@ import useData from "../useData";
 
 const ALIAS_MATERIAS = require("../data/materias.json");
 
+const allCodigos = new Set(
```

Comment:
```suggestion
const CODIGOS = new Set(
```

para que quede claro que es una constante